src/utah-design-system/Sherlock.jsx

```diff
@@ -0,0 +1,603 @@
+    }
+
+    this.outFields = this.getOutFields(
+      null,
+      this.searchField,
+      this.contextField
+    );
+    this.webApi = new WebApi(apiKey, this.signal);
+  }
+
+  async search(searchString) {
+    return await this.webApi.search(this.searchLayer, this.outFields, {
+      predicate: this.getSearchClause(searchString),
+      spatialReference: this.wkid,
+    });
+  }
+
+  async getFeature(searchValue, contextValue) {
+    return await this.webApi.search(
+      this.searchLayer,
+      this.outFields.concat('shape@'),
+      {
+        predicate: this.getFeatureClause(searchValue, contextValue),
+        spatialReference: this.wkid,
+      }
+    );
+  }
+}
+
+class WebApi {
+  constructor(apiKey, signal) {
+    this.baseUrl = 'https://api.mapserv.utah.gov/api/v1/';
+
+    // defaultAttributeStyle: String
+    this.defaultAttributeStyle = 'identical';
+
+    // xhrProvider: dojo/request/* provider
+    //      The current provider as determined by the search function
+    this.xhrProvider = null;
+
+    // Properties to be sent into constructor
+
+    // apiKey: String
+    //      web api key (http://developer.mapserv.utah.gov/AccountAccess)
+    this.apiKey = apiKey;
+
+    this.signal = signal;
+  }
+
+  async search(featureClass, returnValues, options) {
+    // summary:
+    //      search service wrapper (http://api.mapserv.utah.gov/#search)
+    // featureClass: String
+    //      Fully qualified feature class name eg: SGID10.Boundaries.Counties
+    // returnValues: String[]
+    //      A list of attributes to return eg: ['NAME', 'FIPS'].
+    //      To include the geometry use the shape@ token or if you want the
+    //      envelope use the shape@envelope token.
+    // options.predicate: String
+    //      Search criteria for finding specific features in featureClass.
+    //      Any valid ArcObjects where clause will work. If omitted, a TSQL *
+    //      will be used instead. eg: NAME LIKE 'K%'
+    // options.geometry: String (not fully implemented)
+    //      The point geometry used for spatial queries. Points are denoted as
+    //      'point:[x,y]'.
+    // options.spatialReference: Number
+    //      The spatial reference of the input geographic coordinate pair.
+    //      Choose any of the wkid's from the Geographic Coordinate System wkid reference
+    //      or Projected Coordinate System wkid reference. 26912 is the default.
+    // options.tolerance: Number (not implemented)
+    // options.spatialRelation: String (default: 'intersect')
+    // options.buffer: Number
+    //      A distance in meters to buffer the input geometry.
+    //      2000 meters is the maximum buffer.
+    // options.pageSize: Number (not implemented)
+    // options.skip: Number (not implemented)
+    // options.attributeStyle: String (defaults to 'identical')
+    //      Controls the casing of the attributes that are returned.
+    //      Options:
+    //
+    //      'identical': as is in data.
```

Comment:
**question**: since we switched to pg, everything is lower case. I was creating input, upper, and lower in the new api where input would match how you send it in and upper and lower are obvious. But if you're expecting it to be as is in the data, that would simplify things. I believe you're the only one using this hidden feature. thoughts?

Reply:
Yes! I think that we could remove the identical option since it seems irrelevant. b6f0663dc37619229bc028e9c45088865591b4e3